### PR TITLE
adding xz compression for 64bits processors

### PR DESCRIPTION
**Pull request by melodie11**
Here is a diff:
--- a-rebuild.py    2015-10-25 17:00:09.289525950 +0100
+++ b-rebuild.py    2015-10-25 16:59:28.224223838 +0100
@@ -204,7 +204,7 @@
         # Using the branch-call-jump filter provides a compression boost with executable code.
         # This can save a hundred megabytes easily, on an 800MB ISO. The dictionary size must
         # match the block size, and it's advisable to use larger block sizes, like 1MB or 4MB.
-        compression_parameters += ('-Xbcj', 'x86', '-Xdict-size', '100%')
- ```
     compression_parameters += ('-Xbcj', 'x86,ia64', '-Xdict-size', '100%')
  ```
  
   message.sub_info('SquashFS Compression parameters', compression_parameters)
  
   # Create the compressed filesystem

### CODE REVIEW

**Comment by ghost** [comment]
@melodie11 
Thank you for your interest to make a pull request. However, I think you misunderstood something about the xz compression option here. Two things have been clarified below.
1. [This page on Ubuntu Manpage](http://manpages.ubuntu.com/manpages/trusty/man1/xz.1.html) had specified that `x86` option is applicable for both 32-bit or 64-bit x86, which covers most of 64-bit processors used by end-users.
   
   > Different instruction sets have have different alignment:
   > 
   > ```
   >          Filter      Alignment   Notes
   >          x86             1       32-bit or 64-bit x86
   >          PowerPC         4       Big endian only
   >          ARM             4       Little endian only
   >          ARM-Thumb       2       Little endian only
   >          IA-64          16       Big or little endian
   >          SPARC           4       Big or little endian
   > ```
2. [This page on Wikipedia](https://en.wikipedia.org/wiki/IA-64) explains that IA-64 is aimed at servers and high-end machines and covers only Intel-based processor. End-users wouldn't be using this architecture on their PC.
   
   > Itanium is aimed at the enterprise server and high-performance computing (HPC) markets.

Besides above, these are the [differences between IA-64 and X64](http://superuser.com/questions/276994/difference-between-ia64-and-x64) to note. In general, IA-64 is said to be less popular and less supported. Therefore, there is no reason to add `ia64` option in Customizer.

Unless there is any valid reasons to justify this pull request, this pull request should be closed.

**Comment by melodie11** [comment]
Hi,

I had it from the mksquashfs man, as an available option. I didn't know about the xz man
page option.

However, I got different results in first using the default options, then in using the
option with "ia64".

For the details:
I operated in Bento Openbox 14.04 x86_64 (Ubuntu with Openbox), to remix Bento Openbox
14.04.3 x86_64 just updated, with the xz compression on. I didn't change anything else in
the configuration file.

The computer is a Lenovo laptop, T410, with 4 GB RAM and this Intel i5 processor:
Intel(R) Core(TM) i5 CPU       M 520  @ 2.40GHz

cpufreq-info says:
hardware limits: 1.20 GHz - 2.40 GHz
  available frequency steps: 2.40 GHz, 2.40 GHz, 2.27 GHz, 2.13 GHz, 2.00 GHz, 1.87 GHz,
1.73 GHz, 1.60 GHz, 1.47 GHz, 1.33 GHz, 1.20 GHz

and the governor used on each of the 4 cpus is "ondemand".

First rebuilt, using Customizer with the default options:
The original ISO file was 686MB, the resulting file 695MB (no change done in the content,
just pick one, prior done with Ubuntu Builder, check the update, updated Midori –which
weight is very small and not worth mentioning– did apt-get clean, and rebuilt).

Second and third rebuilt, using Customizer with the "ia64" added in the rebuild.py file:
I added the ia64, and rebuilt again but got the same result as the first time, so I
rebooted, restarted Customizer, did a "clean" and remixed again from the first source
ISO : now the resulting ISO is 681MB, which is much closer to what I was expecting, for a
file rebuilt with the zx compression.

Also, I noticed the processors are all working hard while compressing. (I could not use
the tower because the session was hanging, the CPU could not hold it, so I restarted to
work on this laptop, using customizer in text mode, through ssh for more comfort).

What do you think about the two uneven results? Would you want to suggest some additional
tests to do?

Best regards,
Mélodie

On Mon, 26 Oct 2015 02:13:06 -0700
clearkimura notifications@github.com wrote:

> @melodie11 
> Thank you for your interest to make a pull request. However, I think you misunderstood
> something about the xz compression option here. Two things have been clarified below.
> 1. [This page on Ubuntu Manpage](http://manpages.ubuntu.com/manpages/trusty/man1/xz.1.html) had specified that `x86`
>    option is applicable for both 32-bit or 64-bit x86, which covers most of 64-bit
>    processors used by end-users.
>    
>    > Different instruction sets have have different alignment:
>    > 
>    > ```
>    >          Filter      Alignment   Notes
>    >          x86             1       32-bit or 64-bit x86
>    >          PowerPC         4       Big endian only
>    >          ARM             4       Little endian only
>    >          ARM-Thumb       2       Little endian only
>    >          IA-64          16       Big or little endian
>    >          SPARC           4       Big or little endian
>    > ```
> 2. [This page on Wikipedia](https://en.wikipedia.org/wiki/IA-64) explains that IA-64 is
>    aimed at servers and high-end machines and covers only Intel-based processor. End-users
>    wouldn't be using this architecture on their PC.
>    
>    > Itanium is aimed at the enterprise server and high-performance computing (HPC)
>    > markets.
> 
> Besides above, these are the [differences between IA-64 and X64](http://superuser.com/questions/276994/difference-between-ia64-and-x64) to note. In
> general, IA-64 is said to be less popular and less supported. Therefore, there is no
> reason to add `ia64` option in Customizer.
> 
> Unless there is any valid reasons to justify this pull request, this pull request
> should be closed.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/clearkimura/Customizer/pull/114#issuecomment-151069055

## 

JM meets@gmx.fr

**Comment by kamilion** [comment]
I don't think we need to support the old intel itanium series. Nobody used those but banks, I haven't even seen them come in for trade-in for two to three years where I work anymore.

However, I've tried '-Xbcj', 'x86,ia64,arm,armthumb,powerpc,sparc'  before with no ill effects. It will only apply the compression filter if it makes sense to -- and if it increases compression, there's no reason not to use it.

And yes, mksquashfs will use all available cores for compression by default. Unfortunately, everyone has a different amount of processor cores, so we can't really set anything like a 'sane default'... I've got a 64 core machine with 256GB of ram on my test station right now...

**Comment by ghost** [comment]
Thanks to @kamilion for the testing and clarification. From my side, I will add some comments to existing points and from different point of view.

@melodie11 

> What do you think about the two uneven results?
> Would you want to suggest some additional tests to do?

It seems that your testing had uncontrolled factors and the results are therefore, unconvincing. Normally, I'd like to suggest controlled testing procedures, but in this case I'd say "no".

After all, [this manpage of `mkquashfs`](http://manpages.ubuntu.com/manpages/trusty/man1/mksquashfs.1.html) has mentioned what would the `-Xbcj` option do.

> ```
> -Xbcj filter1,filter2,...,filterN
>     Compress  using filter1,filter2,...,filterN in turn (in addition to
>     no filter), and choose the  best  compression.  Available  filters:
>     x86, arm, armthumb, powerpc, sparc, ia64.
> ```

So, if you simply add another filter as `-Xbcj  x86,ia64` like what you suggested in your pull request, in theory, the `mksquashfs` would still prefer `x86`. As the manpage says, it compress and "choose the best compression", [likely] based on your machine architecture.

> [...] remixed again from the first source ISO : now the resulting ISO is 681MB, which is much closer to what I was expecting, for a file rebuilt with the zx compression.

From what I understand, you are trying to reduce the ISO image size the "easy" way. You take an ISO image file, dump it into the remastering tool and rebuild again using `xz` which has better compression ratio than `gzip`. To me, this is "wasteful" way to do it.

You achieve smaller ISO image size, in trade of slower compression speed and slower decompression speed when reading the image later. These drawbacks could be ignored, only if you are working on smaller and modular distribution, such as Slax, when ISO image size is very critical.

In case of mainstream distribution such as Ubuntu and its derivatives, there are much better ways of reducing the ISO image size, however not limited to, such as purging unused locales, unused docs, unused packages and redundant applications.

**Updated**
You can visit [the official Squashfs LZMA/XZ site](http://www.squashfs-lzma.org/) and see their "Compression benchmarks" which uses Slax data. It benefits smaller distributions, but not really suited for mainstream ones.

**Comment by melodie11** [comment]
Hello,

I'll also answer by quoting in the text. I will clarify some points, especially about
"how to reduce the size of the ISO".

Thanks also for your answers, and I hope we can still discuss about some tests to do,
which I would be willing to do.

On Tue, 27 Oct 2015 01:36:19 -0700
clearkimura notifications@github.com wrote:

> Thanks to @kamilion for the testing and clarification. From my side, I will add some
> comments to existing points and from different point of view.

About Itanium, I have read all the pages on the web, that I could, yesterday in the
night, and clearly the processor of my Thinkpad T410 with an Intel i5 M520 does not
belong to the processors listed as Itanium.

This said, it is very unclear to me why the compression results can differ so much from
one rebuild session to another, even when doing a "clean" and then a reboot before
processing a new built.

> @melodie11 
> 
> > What do you think about the two uneven results?
> > Would you want to suggest some additional tests to do?
> 
> It seems that your testing had uncontrolled factors and the results are therefore,
> unconvincing. Normally, I'd like to suggest controlled testing procedures, but in this
> case I'd say "no".

Well, if you change your mind, I'm your girl! (a man would say "I'm your man" :
right? :-) )

> After all, [this manpage of `mkquashfs`](http://manpages.ubuntu.com/manpages/trusty/man1/mksquashfs.1.html) has mentioned what
> would the `-Xbcj` option do.
> 
> > ```
> > -Xbcj filter1,filter2,...,filterN
> >     Compress  using filter1,filter2,...,filterN in turn (in addition to
> >     no filter), and choose the  best  compression.  Available  filters:
> >     x86, arm, armthumb, powerpc, sparc, ia64.
> > ```
> 
> So, if you simply add another filter as `-Xbcj  x86,ia64` like what you suggested in
> your pull request, in theory, the `mksquashfs` would still prefer `x86`. As the manpage
> says, it compress and "choose the best compression", [likely] based on your machine
> architecture.

in theory?
…

> > [...] remixed again from the first source ISO : now the resulting ISO is 681MB, which
> > is much closer to what I was expecting, for a file rebuilt with the zx compression.
> 
> From what I understand, you are trying to reduce the ISO image size the "easy" way. You
> take an ISO image file, dump it into the remastering tool and rebuild again using `xz`
> which has better compression ratio than `gzip`. To me, this is "wasteful" way to do it.
> 
> You achieve smaller ISO image size, in trade of slower compression speed and slower
> decompression speed when reading the image later. These drawbacks could be ignored,
> only if you are working on smaller and modular distribution, such as Slax, when ISO
> image size is very critical.
> 
> In case of mainstream distribution such as Ubuntu and its derivatives, there are much
> better ways of reducing the ISO image size, however not limited to, such as purging
> unused locales, unused docs, unused packages and redundant applications.

Some time ago, I had a discussion on IRC with Kamilion, who showed me one of his build
scripts, where he uses these compression filters, and remembering I had seen it before
while remastering PCLinuxOS, I gave it a try and was pleased to see that it would allow
for a gain up to 20 or 30MB, which can make a real difference whether it would be above or
under a 700MB CD size, 

What comes after is to explain you why I appreciate an added compression filter. About
the compression filters, it adds time for the compression and needs more CPU power than
the gzip option, but it does not add much time for decompression and not much to CPU load
for decompression, as per the squashfs-lzma doc:
http://www.squashfs-lzma.org/

“Compressing to LZMA format is very slow, but this is not a problem as the squashfs
filesystem needs to be compressed only once. Then it's only read million times; so what
we care about is the speed of decompression. LZMA decompression is slower than gzip, but
it's even very fast, about 10-20 MB/s on a 2GHz CPU. ”

There is also a frame presenting different block sizes, and an exemple at the end,
"mksquashfs source1 dest.sqfs -comp xz -b 262144 " which makes me wonder if the block
size chosen currently is the best possible choice. 

@Kamilion : would you know that?

About what I work on:
I don't do mainstream distribution remixing, but start almost from the ground up, using
Ubuntu Mini Remix, a small ISO provided by Fabrizio Balliani, which purpose is to build
whatever kind of Ubuntu Remix one wants. It's very small and made with the base
meta-packages.
http://www.ubuntu-mini-remix.org/

On the right side, in the "Projects based on UMR" section, you can see "Bento", this is
what I do. A version with an Openbox branded desktop to be easy for end users. So I choose
carefully the programs I want to include, and I'm still climbing the learning curve for
some parts to be improved, but I can assure you the compression filters are the ultimate
method I'm to use in order to reduce the size of the distribution: the ISOs are online,
and there is no way I'll remove the docs, or the locales. 

I use them IRL around me to migrate some users from Windows. For the ISOs on the web, I
want the locales and the docs to be available for everyone. The so-called docs contain
mostly the information about the packages, such as who are the authors, I want to
respect that. The locales, well, it's obvious: if the locales are removed, a Spanish
user will have to reinstall each and every package if he wants to use the distro in his
language, and same for potential users from other countries. 

I can also say it's easy to reuse Bento Openbox Remix to create a new setup and rebuild
from it or else. 

Starting 2012, I used Ubuntu-Builder, which seemed the easiest way for me to start
remixing, and although the development of the program was halted last year in March, I
was still using it. Kamilion told me about your work on Customizer, and I started trying
it just a pair of days ago. 

Issues met with Customizer:
At first, it would not start, neither from the menus nor from
the console. 

Kamilion tought me two things: 
- that pkexec does not start Customizer in Openbox and in Lubuntu:
- how to turn the debug on, and from there the console told
  me that a filesystem.manifest-remove was missing in the casper directory. This is because
  UMR replaced filesystem.manifest-desktop starting from Vivid only, and I'm working on the
  Trusty LTS. So I created that file (using the result of a diff between the manifest and
  the desktop manifest) and the problem was solved. 

Then when I used it in my tower, it froze the desktop and the logs showed that the CPU's
could not handle it, which led me to use the T410 instead. (The tower has a 'AMD Athlon
(tm) II X2 260 Processor').

Ok, Customizer gui needs more CPU power than Ubuntu-Builder does. (But on the other hand,
it does not kill dbus, which Ubuntu-Builder leaves impossible to restart after each
build).

In summary: 
I rebuilt the 2 ISOs Bento Openbox Trusty with an added filesystem.manifest-remove file
in the casper directory, using Ubuntu-Builder;

Then I renamed the two ISOs in a way I know they are meant to be rebuilt with Customizer
(for the tests purpose before starting to use Customizer currently);

Then I observed the differences between the size of the source and the final ISO. I also
installed each final ISO in Virtualbox to make sure it works right, which it does.

So if you change your mind, please let me know how I can test and benchmark the
LZMA2 compression filters? (I could use the UMR version Wily with very little addition in
it, well chosen, to start producing Bento Sushi Wily, for exemple).

Best regards,
Mélodie

**Comment by ghost** [comment]
@melodie11 
That was very long text to digest... I guess I'm always being mistaken for the developer, as long as I'm the owner of this repo. Sorry to dissappoint you with my naive replies.

**Important first**

> Kamilion told me about your work on Customizer, and I started trying it just a pair of days ago.

Customizer is very far from being my work; I've been just a happened-to-be contributor. Should there be any misunderstanding between us, including @kamilion also, I'd like to make this clear: I'm _not_ the developer of Customizer. You can read the background story in issue #70 later.

> [...] clearly the processor of my Thinkpad T410 with an Intel i5 M520 does not belong to the processors listed as Itanium.

So I assume this as indirect way of saying that this pull request is indeed invalid, and therefore will be closed after this comment.

Now that you have been informed that I'm _not_ the developer, if you still want to read my remaining replies, feel free to do so. But my answers will likely _not_ meet your expectations.

**The remaining**

> This said, it is very unclear to me why the compression results can differ so much from one rebuild session to another, even when doing a "clean" and then a reboot before processing a new built.

I can't see how the testing you did as "clean". If those testing were indeed "clean", you wouldn't get different results when you repeat the testing. It sounds clear to me that either you have not tested in same manner, or you failed to control some factors while working on your remix.

> >  So, if you simply add another filter as `-Xbcj  x86,ia64` like what you suggested in
> > your pull request, in theory, the `mksquashfs` would still prefer `x86`. As the manpage
> > says, it compress and "choose the best compression", [likely] based on your machine
> > architecture.
> 
> in theory?
> …

I can't claim what I'd written as a fact, since I didn't do any testing for clarification. It could be wrong. Hence, "in theory" from my saying. Perhaps @kamilion is the better person to answer "in practice" based on experience.

> What comes after is to explain you why I appreciate an added compression filter. About the compression filters, it adds time for the compression and needs more CPU power than the gzip option, [...]

According to the `mksquashfs` manpage, the "filters" are referring to the architectures (`x86 ia64` etc.), and _not_ the compression format. Quoted again below.

> -Xbcj filter1,filter2,...,filterN
> Compress using filter1,filter2,...,filterN in turn (in addition to no filter), and choose the best compression. Available filters: x86, arm, armthumb, powerpc, sparc, ia64.

That being said, I'm confused and lost by what you had said earlier.

> I don't do mainstream distribution remixing, but start almost from the ground up, using Ubuntu Mini Remix, [...]

While I have indeed said "In case of mainstream distribution such as Ubuntu and its derivatives [...]", I was referring to your testing case, which you had loaded Bento Openbox (Ubuntu with Openbox), with size of "The original ISO file was 686MB".

> Issues met with Customizer:
> At first, it would not start, neither from the menus nor from the console.
> 
> Kamilion tought me two things:
> - that pkexec does not start Customizer in Openbox and in Lubuntu:
> - how to turn the debug on, and from there the console told [...]

This is too much for me to accept at one time... This is an issue by itself, which I encounter from time to time, even before I became the repo owner. That is, nobody bothers to give useful feedback and let _other users_ know what works and what didn't work. Give me a break, please.

> [...] a filesystem.manifest-remove was missing in the casper directory. This is because UMR replaced filesystem.manifest-desktop starting from Vivid only, and I'm working on the Trusty LTS.

I'm certainly not aware, because I have only tested up to Trusty releases. And it would be the last releases to be tested on my side. Wish all the best to future contributors for testing newer releases.

> Ok, Customizer gui needs more CPU power than Ubuntu-Builder does. (But on the other hand, it does not kill dbus, which Ubuntu-Builder leaves impossible to restart after each build).

You are comparing an apple to an orange. Customizer GUI is now written in Python and Ubuntu Builder was written in Gambas (being a fork of the deprecated Customizer 3.x).

> So if you change your mind, please let me know how I can test and benchmark the LZMA2 compression filters? (I could use the UMR version Wily with very little addition in it, well chosen, to start producing Bento Sushi Wily, for exemple).

Nothing useful to be advised from my side, being the end-user.

**Comment by melodie11** [comment]
Hi,

"clean" as in :

---

# customizer -h:

  -t, --clean    Clean all temporary files and folders

---

Steps related to "clean between the two builds":

"customizer -t > reboot > check /etc/customizer.conf > customizer -e > customizer -c >
apt-get update && apt-get dist-upgrade > exit > customizer -r".

theory or not theory:
Kamilion lives 8 hours earlier than I do, on the planet, and we had a talk today, he said
he is going to do some tests related to compression in one of his machines, today. Now
here, it's 6pm, while it's 10 am for him, so I will be looking forward for his results in
the coming days. And yes, I agree with you, he is the right person for tests in practice.

About this part:

> This is too much for me to accept at one time... This is an issue by itself, which I
> encounter from time to time, even before I became the repo owner. That is, nobody
> bothers to give useful feedback and let _other users_ know what works and what didn't
> work. Give me a break, please.

I never bother to give unuseful feedback, and always try to provide useful feedback,
depending on my skills in the matter. What does "give me a break" mean? I'm French (south
France), and I don't always understand English right. So this sounds rough on the edges
for me: was it?

Besides that, I didn't know at all that you are not the developer, as you are the one who
answered the pull request. So sorry if I have hurted you in any way or bothered, this was
not my intention. 

Best regards,
Mélodie

On Tue, 27 Oct 2015 09:49:45 -0700
clearkimura notifications@github.com wrote:

> @melodie11 
> That was very long text to digest... I guess I'm always being mistaken for the
> developer, as long as I'm the owner of this repo. Sorry to dissappoint you with my
> naive replies.
> 
> **Important first**
> 
> > Kamilion told me about your work on Customizer, and I started trying it just a pair
> > of days ago.
> 
> Customizer is very far from being my work; I've been just a happened-to-be contributor.
> Should there be any misunderstanding between us, including @kamilion also, I'd like to
> make this clear: I'm _not_ the developer of Customizer. You can read the background
> story in issue #70 later.
> 
> > [...] clearly the processor of my Thinkpad T410 with an Intel i5 M520 does not belong
> > to the processors listed as Itanium.
> 
> So I assume this as indirect way of saying that this pull request is indeed invalid,
> and therefore will be closed after this comment.
> 
> Now that you have been informed that I'm _not_ the developer, if you still want to read
> my remaining replies, feel free to do so. But my answers will likely _not_ meet your
> expectations.
> 
> **The remaining**
> 
> > This said, it is very unclear to me why the compression results can differ so much
> > from one rebuild session to another, even when doing a "clean" and then a reboot
> > before processing a new built.
> 
> I can't see how the testing you did as "clean". If those testing were indeed "clean",
> you wouldn't get different results when you repeat the testing. It sounds clear to me
> that either you have not tested in same manner, or you failed to control some factors
> while working on your remix.
> 
> > >  So, if you simply add another filter as `-Xbcj  x86,ia64` like what you suggested
> > > in your pull request, in theory, the `mksquashfs` would still prefer `x86`. As the
> > > manpage says, it compress and "choose the best compression", [likely] based on
> > > your machine architecture.
> > 
> > in theory?
> > …
> 
> I can't claim what I'd written as a fact, since I didn't do any testing for
> clarification. It could be wrong. Hence, "in theory" from my saying. Perhaps @kamilion
> is the better person to answer "in practice" based on experience.
> 
> > What comes after is to explain you why I appreciate an added compression filter.
> > About the compression filters, it adds time for the compression and needs more CPU
> > power than the gzip option, [...]
> 
> According to the `mksquashfs` manpage, the "filters" are referring to the architectures
> (`x86 ia64` etc.), and _not_ the compression format. Quoted again below.
> 
> > -Xbcj filter1,filter2,...,filterN
> > Compress using filter1,filter2,...,filterN in turn (in addition to no filter), and
> > choose the best compression. Available filters: x86, arm, armthumb, powerpc, sparc,
> > ia64.
> 
> That being said, I'm confused and lost by what you had said earlier.
> 
> > I don't do mainstream distribution remixing, but start almost from the ground up,
> > using Ubuntu Mini Remix, [...]
> 
> While I have indeed said "In case of mainstream distribution such as Ubuntu and its
> derivatives [...]", I was referring to your testing case, which you had loaded Bento
> Openbox (Ubuntu with Openbox), with size of "The original ISO file was 686MB".
> 
> > Issues met with Customizer:
> > At first, it would not start, neither from the menus nor from the console.
> > 
> > Kamilion tought me two things:
> > - that pkexec does not start Customizer in Openbox and in Lubuntu:
> > - how to turn the debug on, and from there the console told [...]
> 
> This is too much for me to accept at one time... This is an issue by itself, which I
> encounter from time to time, even before I became the repo owner. That is, nobody
> bothers to give useful feedback and let _other users_ know what works and what didn't
> work. Give me a break, please.
> 
> > [...] a filesystem.manifest-remove was missing in the casper directory. This is
> > because UMR replaced filesystem.manifest-desktop starting from Vivid only, and I'm
> > working on the Trusty LTS.
> 
> I'm certainly not aware, because I have only tested up to Trusty releases. And it would
> be the last releases to be tested on my side. Wish all the best to future contributors
> for testing newer releases.
> 
> > Ok, Customizer gui needs more CPU power than Ubuntu-Builder does. (But on the other
> > hand, it does not kill dbus, which Ubuntu-Builder leaves impossible to restart after
> > each build).
> 
> You are comparing an apple to an orange. Customizer GUI is now written in Python and
> Ubuntu Builder was written in Gambas (being a fork of the deprecated Customizer 3.x).
> 
> > So if you change your mind, please let me know how I can test and benchmark the LZMA2
> > compression filters? (I could use the UMR version Wily with very little addition in
> > it, well chosen, to start producing Bento Sushi Wily, for exemple).
> 
> Nothing useful to be advised from my side, being the end-user.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/clearkimura/Customizer/pull/114#issuecomment-151565548

## 

JM meets@gmx.fr

**Comment by ghost** [comment]
There were many vague points throughout the "off-topic" discussions, and more towards ignoring the main intention of this pull request. Summarized points below.
- issues with Customizer should have been created via "New Issue"
- this is _not_ a "Questions & Answers" site
- nobody has much time to read such lengthy text (no thanks for stealing their time)

Enough said.

**Comment by melodie11** [comment]
> ```
> nobody has much time to read such lengthy text (no thanks for stealing their time)
> ```

sorry to have to answer such a disagreeable reading, but, it's on you. Next time you don't want lengthy answers, don't start with your kind advice which are already mentioned in the documentation. Referring to the "how to reduce size in an iso", thus will also save my time. 

This is ending the discussion on my side.
